### PR TITLE
Have ParserWithLocation report comments

### LINKED ISSUE #94: Comments
Provide a way to optionally parse comments

### PR DESCRIPTION
**Pull request by bakkot**
This is a straight port of the similar change to shift-parser-js.

Fixes #94.

### CODE REVIEW

File: src/main/java/com/shapesecurity/shift/es2016/parser/ParserWithLocation.java [hunk 1]
```diff
@@ -34,6 +59,11 @@ public Module parseModule(@Nonnull String text) throws JsError {
 		return this.locations.get(node);
 	}
 
+	@Nonnull
+	public ImmutableList<Comment> getComments() {
+		return this.comments.reverse();
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/shift/es2016/parser/ParserWithLocation.java:63]: Why does the order matter? If they want it reversed, they can reverse it.

Comment (bakkot) [reply src/main/java/com/shapesecurity/shift/es2016/parser/ParserWithLocation.java:63]: Consistency with shift-parser-js, mainly.

**Review by michaelficarra — APPROVED** [review]
LGTM otherwise.

File: src/main/java/com/shapesecurity/shift/es2016/parser/ParserWithLocation.java (hunk 1 above)
Comment (michaelficarra) [reply src/main/java/com/shapesecurity/shift/es2016/parser/ParserWithLocation.java:63]: I would say that order just isn't guaranteed. A responsible consumer would be sorting them by offset anyway. Nobody is promising them anything about order.

Comment (bakkot) [reply src/main/java/com/shapesecurity/shift/es2016/parser/ParserWithLocation.java:63]: If we expect responsible consumers to be sorting them by offset, we should just sort them by offset ourselves, and guarantee that. Which is the current behavior.

Comment (michaelficarra) [reply src/main/java/com/shapesecurity/shift/es2016/parser/ParserWithLocation.java:63]: I mean a consumer who wants them sorted. There's plenty of consumers who won't care about ordering.

**Pull request merged**